samples/cpp/videocapture_with_audio.cpp

```diff
@@ -0,0 +1,146 @@
+                 "q,Q,esc -- quit\n"
+                 "You can also pass the path to an image sequence and OpenCV will treat the sequence just like a video\n"
+                 "example: --input=right%02d.jpg\n");
+
+    // Check for 'help' argument to display usage information
+    if (parser.has("help"))
+    {
+        parser.printMessage();
+        return 0;
+    }
+
+    // Get input file path and check if audio extraction is required
+    string file = parser.get<string>("input");
+    bool isWithAudio = parser.has("with_audio");
+
+    Mat videoFrame;
+    Mat audioFrame;
+    vector<vector<Mat>> audioData;
+    VideoCapture cap;
+
+    // Setup VideoCapture based on user input
+    if (isWithAudio) {
+        if (file.empty()) {
+            // No input file, extract audio from default device
+            cout << "Extracting audio samples from default device..." << endl;
+            vector<int> micParams{ CAP_PROP_AUDIO_STREAM, 0, CAP_PROP_VIDEO_STREAM, -1, CAP_PROP_AUDIO_DATA_DEPTH, CV_16S };
+            cap.open(0, CAP_ANY, micParams);
+            if (!cap.isOpened()) {
+                cerr << "ERROR! Unable to open camera and microphone" << endl;
+                return -1;
+            }
+        } else {
+            // Open input file for both video and audio
+            vector<int> fileParams{ CAP_PROP_AUDIO_STREAM, 0, CAP_PROP_VIDEO_STREAM, 0, CAP_PROP_AUDIO_DATA_DEPTH, CV_16S };
+            cap.open(file, CAP_ANY, fileParams);
+            if (!cap.isOpened()) {
+                cerr << "ERROR! Unable to open file: " + file << endl;
+                return -1;
+            }
+        }
+    } else {
+        // Setup VideoCapture for video only
+        if (file.empty()) {
+            // No input file, capture video from default camera
+            cap.open(0, CAP_ANY);
+            if (!cap.isOpened()) {
+                cerr << "ERROR! Unable to open camera" << endl;
+                return -1;
+            }
+        } else {
+            // Open input video file
+            cap.open(file, CAP_ANY);
+            if (!cap.isOpened()) {
+                cerr << "ERROR! Unable to open file: " + file << endl;
+                return -1;
+            }
+        }
+    }
+
+    // Audio settings and info
+    const int audioBaseIndex = (int)cap.get(CAP_PROP_AUDIO_BASE_INDEX);
+    const int numberOfChannels = (int)cap.get(CAP_PROP_AUDIO_TOTAL_CHANNELS);
+    cout << "Audio Data Depth: " << depthToString((int)cap.get(CAP_PROP_AUDIO_DATA_DEPTH)) << endl;
+    cout << "Audio Samples Per Second: " << cap.get(CAP_PROP_AUDIO_SAMPLES_PER_SECOND) << endl;
+    cout << "Total Audio Channels: " << cap.get(CAP_PROP_AUDIO_TOTAL_CHANNELS) << endl;
+    cout << "Total Audio Streams: " << cap.get(CAP_PROP_AUDIO_TOTAL_STREAMS) << endl;
+
+    int numberOfFrames = 0;
+    int numberOfSamples = 0;
+    audioData.resize(numberOfChannels);
```

Comment:
@asmorkalov This solved the seg fault
```suggestion
    if (numberOfChannels > 0){
        audioData.resize(numberOfChannels);
    }
```